### PR TITLE
Overmap 4.8: Asteroid Fields

### PR DESCRIPTION
**Pull request by goober3**
## About The Pull Request

image

large asteroids now use multibiome planet generation.

asteroids generate in fields of small to large asteroids, with space carp spawning in the space between them.

space carp have been added to the mining faction, to avoid turf wars between carp dens & other dens

## Why It's Good For The Game

large asteroids were large, featureless rocks packed with mobs and resources. asteroid fields are infinitely cooler.

## Changelog

:cl:
tweak: Large asteroids have been reworked into asteroid fields.
/:cl:

### CODE REVIEW

**Review by MarkSuckerberg — APPROVED** [review]
thank you. the asteroid mapgen was basically me going "how basic can I make this so that it exists" and then legitimately just making a circle with a bit of randomness

although you can probably just delete the old file honestly it won't be re-used I can promise you that

**Comment by thgvr** [comment]
Do you think you could remove goliaths from the spawn pools of these? Maybe add a rare spawn of space sharks to go with the carp? The hivelords are good, though.

File: code/modules/mob/living/simple_animal/hostile/carp.dm [hunk 1]
```diff
@@ -40,7 +40,7 @@
 	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
 	minbodytemp = 0
 	maxbodytemp = 1500
-	faction = list("carp")
+	faction = list("carp, mining")
```
Comment (Erikafox) [inline code/modules/mob/living/simple_animal/hostile/carp.dm]:
```suggestion
	faction = list("carp", "mining")
```

File: code/game/objects/structures/spawner.dm [hunk 2]
```diff
@@ -147,3 +147,8 @@ GLOBAL_LIST_INIT(astroloot, list(
 	name = "wumborian fugu den"
 	desc = "A den housing a nest of wumborian fugus, how do they all even fit in there?"
 	mob_types = list(/mob/living/simple_animal/hostile/asteroid/fugu)
+
+/obj/structure/spawner/mining/carp
+	name = "carp den"
+	desc = "A den housing a nest of space carp, seems fishy!"
+	mob_types = list(/mob/living/simple_animal/hostile/carp)
```
Comment (Erikafox) [inline code/game/objects/structures/spawner.dm:155]: please look at making some presets (like how other planets do their spawners)|
Also note you can add a custom spawn sound

File: code/datums/mapgen/planetary/AsteroidGenerator.dm [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+	feature_spawn_list = list(
+	/obj/structure/geyser/random = 1
+	)
+
+	mob_spawn_list = list(
+	/mob/living/simple_animal/hostile/asteroid/goliath = 25, /obj/structure/spawner/mining/goliath = 30, \
+		/mob/living/simple_animal/hostile/asteroid/basilisk = 25, /obj/structure/spawner/mining = 30, \
+		/mob/living/simple_animal/hostile/asteroid/hivelord = 25, /obj/structure/spawner/mining/hivelord = 30, \
+		/mob/living/simple_animal/hostile/asteroid/goldgrub = 10
```
Comment (Erikafox) [inline code/datums/mapgen/planetary/AsteroidGenerator.dm]:
```suggestion
		 /obj/structure/geyser/random = 1,
		 /obj/structure/spawner/mining/goliath = 1,
		 /obj/structure/spawner/mining = 1,
		 /obj/structure/spawner/mining/hivelord = 1
	)

	mob_spawn_list = list(
		/mob/living/simple_animal/hostile/asteroid/goliath = 25, 
		/mob/living/simple_animal/hostile/asteroid/basilisk = 25, 
		/mob/living/simple_animal/hostile/asteroid/hivelord = 25, 
		/mob/living/simple_animal/hostile/asteroid/goldgrub = 10
```

See lower comment on making the spawner presets for regions. Have the spawners be features so they don't stack up oppressively

File: code/datums/mapgen/planetary/AsteroidGenerator.dm [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+	)
+	open_turf_types = list(
+	/turf/open/floor/plating/asteroid/airless = 1
+	)
+
+/datum/biome/cave/asteroid/vanilla
+	flora_spawn_list = list(
+	/obj/structure/flora/ash/space/voidmelon = 1,
+	/obj/structure/flora/rock = 1,
+	/obj/structure/flora/rock/pile = 1
```
Comment (Erikafox) [inline code/datums/mapgen/planetary/AsteroidGenerator.dm]: indent 1 tab

File: code/datums/mapgen/planetary/AsteroidGenerator.dm [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+	open_turf_types = list(
+	/turf/open/floor/plating/ice/airless = 1
+	)
+
+	mob_spawn_list = list(
+	/mob/living/simple_animal/hostile/asteroid/goliath = 25, \
+		/mob/living/simple_animal/hostile/asteroid/basilisk = 25, \
+		/mob/living/simple_animal/hostile/asteroid/hivelord = 25, \
+		/mob/living/simple_animal/hostile/asteroid/goldgrub = 10
+		)
```
Comment (Erikafox) [inline code/datums/mapgen/planetary/AsteroidGenerator.dm]:
```suggestion
	mob_spawn_list = list(
		/mob/living/simple_animal/hostile/asteroid/goliath = 25, 
		/mob/living/simple_animal/hostile/asteroid/basilisk = 25, 
		/mob/living/simple_animal/hostile/asteroid/hivelord = 25, 
		/mob/living/simple_animal/hostile/asteroid/goldgrub = 10
	)
```

File: code/datums/mapgen/planetary/AsteroidGenerator.dm [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+	flora_spawn_chance = 2
+	feature_spawn_chance = 1
+	mob_spawn_chance = 6
+
+/datum/biome/cave/asteroid/ice
+	open_turf_types = list(
+	/turf/open/floor/plating/ice/airless = 1
```
Comment (Erikafox) [inline code/datums/mapgen/planetary/AsteroidGenerator.dm]:
```suggestion
		/turf/open/floor/plating/ice/airless = 1
```

File: code/datums/mapgen/planetary/AsteroidGenerator.dm [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+	)
+
+/datum/biome/asteroid
+	open_turf_types = list(
+	/turf/open/space = 10
+	)
+
+/datum/biome/asteroid/carp
+	mob_spawn_list = list(
+	/mob/living/simple_animal/hostile/carp = 1
```
Comment (Erikafox) [inline code/datums/mapgen/planetary/AsteroidGenerator.dm]:
```suggestion
		/mob/living/simple_animal/hostile/carp = 1
```

File: code/datums/mapgen/planetary/AsteroidGenerator.dm [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+			BIOME_LOW_HUMIDITY = /datum/biome/cave/asteroid/vanilla,
+			BIOME_MEDIUM_HUMIDITY = /datum/biome/cave/asteroid/carp_den,
+			BIOME_HIGH_HUMIDITY = /datum/biome/cave/asteroid/carp_den,
+			BIOME_HIGHEST_HUMIDITY = /datum/biome/cave/asteroid/carp_den
+		)
+	)
+
+/datum/biome/asteroid
+	open_turf_types = list(
+	/turf/open/space = 10
```
Comment (Erikafox) [inline code/datums/mapgen/planetary/AsteroidGenerator.dm]:
```suggestion
		/turf/open/space = 10
```

File: code/datums/mapgen/planetary/AsteroidGenerator.dm [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+		/mob/living/simple_animal/hostile/asteroid/basilisk = 25, \
+		/mob/living/simple_animal/hostile/asteroid/hivelord = 25, \
+		/mob/living/simple_animal/hostile/asteroid/goldgrub = 10
+		)
+
+	mob_spawn_chance = 2
+
+/datum/biome/cave/asteroid/carp_den
+	closed_turf_types =  list(
+	/turf/closed/mineral/random = 5
```
Comment (Erikafox) [inline code/datums/mapgen/planetary/AsteroidGenerator.dm]:
```suggestion
		/turf/closed/mineral/random = 5
```

File: code/datums/mapgen/planetary/AsteroidGenerator.dm [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+	)
+	open_turf_types = list(
+	/turf/open/floor/plating/asteroid/airless = 1
```
Comment (Erikafox) [inline code/datums/mapgen/planetary/AsteroidGenerator.dm]:
```suggestion
		/turf/open/floor/plating/asteroid/airless = 1
```

File: code/datums/mapgen/planetary/AsteroidGenerator.dm [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+	)
+
+	flora_spawn_list = list(
+	/obj/structure/flora/ash/space/voidmelon = 3,
+	/obj/structure/flora/rock = 1,
+	/obj/structure/flora/rock/pile = 1
```
Comment (Erikafox) [inline code/datums/mapgen/planetary/AsteroidGenerator.dm]:
```suggestion
		/obj/structure/flora/ash/space/voidmelon = 3,
		/obj/structure/flora/rock = 1,
		/obj/structure/flora/rock/pile = 1
```

File: code/datums/mapgen/planetary/AsteroidGenerator.dm [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+	)
+
+	feature_spawn_list = list(
+	/obj/structure/geyser/random = 1
```
Comment (Erikafox) [inline code/datums/mapgen/planetary/AsteroidGenerator.dm]:
```suggestion
		/obj/structure/geyser/random = 1
```

File: code/datums/mapgen/planetary/AsteroidGenerator.dm [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+	)
+
+	mob_spawn_list = list(
+	/mob/living/simple_animal/hostile/carp = 25, /obj/structure/spawner/mining/carp = 30, \
+		/mob/living/simple_animal/hostile/carp/megacarp = 30
```
Comment (Erikafox) [inline code/datums/mapgen/planetary/AsteroidGenerator.dm]:
```suggestion
		/mob/living/simple_animal/hostile/carp = 25,
 		/obj/structure/spawner/mining/carp = 30, 
		/mob/living/simple_animal/hostile/carp/megacarp = 30
```
Although spawners should be features.

**Review by Erikafox — COMMENTED** [review]
mostly style and comments on what you *could* do

**Comment by goober3** [comment]
ough

**Pull request merged**